.pre-commit-config.yaml

```diff
@@ -53,6 +53,11 @@ repos:
         # Replace --quiet with --verbose to debug issues.
         - --quiet
 
+  - repo: https://github.com/python-poetry/poetry
+    rev: 2.0.1
+    hooks:
+      - id: poetry-lock
```

Comment:
lock or check?  I think it's the same dilemma with build-schema?

Reply:
According to poetry docs:
> The poetry-lock hook calls the poetry lock command to make sure the lock file is up-to-date when committing changes.
There is also `poetry-check` but I don't know if we need it right now

Reply:
I think we can start with check and then allow developer to manually update - the same as `pnpm install --frozen-lockfile` fails without update 🤔